week1/index.js

```diff
@@ -1,10 +1,86 @@
-// To run this file: node index.js
-// Then in your browser: http://localhost:8080
-var http = require('http');
+// Declare global variables
+const http = require('http');
+//start server 
+const server = http.createServer();
+//This port was already chosen by the assignment maker.
+const port = 8080;
+//Object containing the respons text and the content type.
+//this response is the basic response.
+//it will be passed around by functions and modified untils it's in shape to be returned to the client
+let responsToSendBack = {
+    code:'<html></html>',
+    type:'text/html',
+    statusCode: 200,
+    state: Number
+};
 
-var port = 8080;
+//the original state which will always be 10. 
+//not allowed to change so it's a constant
+const originalState = 10;
 
-var server = http.createServer();
+//assign the begin state to the response object in order to modify it based on the users wish
+responsToSendBack.state = originalState;
+
+
+
+// Function that checks which URL input we got from the user.
+//Initiates the right action based on the user choice
+//takes two parameters: the url the user typed and the responseToSendBack so we can start modifying it
+function checkInput (url,data) {
+    switch (url) {
+        case '/state':
+            data = stateResponsHtml(data);
+            break;
+        case '/add':
+            data.state = addOne(data.state);
+            data = operationsResponseHtml(data);
+            break;
+        case '/remove':
+            data.state = subOne(data.state);
+            data = operationsResponseHtml(data);
+            break;
+        case '/reset':
+            data.state = originalState;
+            data = operationsResponseHtml(data);
+            break;
+        default:
+            otherUrls(data);
+    }
+    return data;
+}
+
+//Function to increase the state by 1
+function addOne (input){
+    return input += 1;
+};
+//Function to decrease the state by 1
+function subOne (input) {
+    return input -= 1;
+};
+
+//Function to build the response HTML for subtraction and addition
+//takes the whole response object and return the modified version.
```

Comment:
This is a great function 'docstring', which means a comment about what a function does. It explains what the function will do, what it takes in and what it returns out. You can also leave comments about why you implemented things in a certain way.